lib/charms/opensearch/v0/opensearch_nodes_exclusions.py

```diff
@@ -70,55 +72,134 @@ def delete_current(self) -> None:
                 self._scope, ALLOCS_TO_DELETE, ",".join(current_allocations)
             )
 
+    def _removed_units_to_cleanup(self) -> Optional[List[str]]:
+        """Deletes all units that have left the cluster via Juju.
+
+        This method ensures we keep a small list of voting exclusions at all times.
+        """
+        deployment_desc = self._charm.opensearch_peer_cm.deployment_desc()
+        if not deployment_desc or deployment_desc.typ not in [
+            DeploymentType.MAIN_ORCHESTRATOR,
+            DeploymentType.FAILOVER_ORCHESTRATOR,
+        ]:
+            return []
+
+        short_id = self._charm.opensearch_peer_cm.deployment_desc().app.short_id
+        # We do not need to add our own unit as this code is running on it!
+        peers = self._charm.model.get_relation(PeerRelationName)
+        peers = [] if not peers else peers.units
+
+        cms = set(
+            [
+                unit.name + "." + short_id
+                for relation in self._charm.model.relations.get(
+                    PeerClusterOrchestratorRelationName
+                )
+                for unit in relation.units
+            ]
+            + [unit.name + "." + short_id for unit in peers]
+        )
```

Comment:
I believe you can replace the above with: 
1. Check if current is main provider `if self.charm.opensearch_peer_cm.is_provider(typ="main")`
    1. fetch all `units = self.charm.peers_data.get(Scope.APP, "cluster_fleet_apps")`
    2. parse it as show [here](https://github.com/canonical/opensearch-operator/blob/main/lib/charms/opensearch/v0/opensearch_peer_clusters.py#L401)
2. else => small deployments
    1. `units = [format_unit_name(u) for u in all_units(self.charm)]`